### PR TITLE
Message Queue Storage

### PR DESCRIPTION
**Pull request by jiivan**
Complete API for message queue storage.

part of #2223

### CODE REVIEW

File: golem/database/schemas/024_create_message_queue.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+def migrate(migrator, database, fake=False, **kwargs):
+    @migrator.create_model  # pylint: disable=unused-variable
+    class QueuedMessage(pw.Model):
+        node = pw.CharField()
+        msg_version = pw.CharField()
+        msg_cls = pw.CharField()
+        msg_data = pw.BlobField()
+        created_date = pw.DateTimeField(default=datetime.datetime.now)
+        modified_date = pw.DateTimeField(default=datetime.datetime.now)
+
```
Comment (shadeofblue) [inline golem/database/schemas/024_create_message_queue.py:19]: will messages live forever if they cannot be sent? maybe they should have some lifetime/timeout set? possibly depending on a type maybe? ... (unsure)

File: golem/model.py [hunk 2]
```diff
@@ -515,6 +535,51 @@ def as_message(self) -> message.base.Message:
...
+        message.base.verify_version(str(self.msg_version))
+        msg = golem_messages.load(
+            self.msg_data,
+            None,
+            None,
+            check_time=False,
+        )
+        msg.header = msg_dt.MessageHeader(
+            msg.header[0],
+            int(time.time()),
```
Comment (shadeofblue) [inline golem/model.py:567]: are we sure that this always produces the correct unix epoch?

File: golem/network/transport/msg_queue.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            except msg_exceptions.MessageError:
+                logger.info(
+                    'Invalid message in queue.'
+                    ' db_model=%s',
+                    db_model,
+                    exc_info=True,
+                )
+                continue
+            finally:
+                db_model.delete_instance()
```
Comment (shadeofblue) [inline golem/network/transport/msg_queue.py:57]: hmm... what if sending of a message fails at the time it's performed? will message be returned to the queue? ... shouldn't a message be removed from the queue only after it has been successfully sent? ...

File: golem/model.py (hunk 2 above)
Comment (jiivan) [reply golem/model.py:567]: Only case it didn't produce expected results was on AppVeyor. We don't use AppVeyor anymore.

File: golem/database/schemas/024_create_message_queue.py (hunk 1 above)
Comment (kmazurek) [reply golem/database/schemas/024_create_message_queue.py:19]: I think these are removed after `MESSAGE_QUEUE_MAX_AGE`, see method `sweep` in `msg_queue.py`.

Comment (shadeofblue) [reply golem/database/schemas/024_create_message_queue.py:19]: yeah, I noticed later

File: golem/network/transport/msg_queue.py (hunk 3 above)
Comment (jiivan) [reply golem/network/transport/msg_queue.py:57]: IMHO the cost of implementing this would be higher than the benefits.

**Review by kmazurek — APPROVED** [review]
LGTM

File: golem/network/transport/msg_queue.py (hunk 3 above)
Comment (shadeofblue) [reply golem/network/transport/msg_queue.py:57]: okay, we may leave that as a future improvement if it deems necessary...

**Review by shadeofblue — APPROVED** [review]
okay, as mentioned above, we can leave it as a future improvement

**Pull request merged**